File: C++/ЕГЭ2021;2022/24/Решение заданий. Поляков/4919.cpp
#include<bits/stdc++.h>
#include<regex>
using namespace std;
int main(){
    fstream f; f.open("191.txt");
    string s; f >> s; 
    s=regex_replace(s, regex("B"), "B B");
    s=regex_replace(s, regex("A"), "A A");
    istringstream iss(s);

    vector<string>str(istream_iterator<string>{iss}, istream_iterator<string>());
    int res=0;
    for(auto&e:str){
        if(count(e.begin(), e.end(), 'F')==2 && e.size()>=20 && e[0]=='A' && e.back()=='B'){
            res++;
        }
    }
    cout << res;
}

File: Python/ЕГЭ2021;2022/24/Решение заданий. Поляков/4919.py
'''
Текстовый файл 24-191.txt содержит строку из заглавных латинских букв, всего не более чем 
из 106 символов. Определите количество подстрок длиной не менее 20 символов, которые начинаются 
буквой A, содержат ровно две буквы F, заканчиваются буквой B и не содержат других букв A и B, 
кроме первой и последней.
'''
# https://prnt.sc/1MpRHtwYgWKy


f = open('24.txt')
s = f.readline()


s = s.replace('A', 'A A', s.count('A')).replace('B', 'B B', s.count('B'))


array = [x for x in s.split() if (x[0] == 'A') and (x[len(x) - 1] == 'B') and (x.count('F')
         == 2) and (x.count('A') == 1) and (x.count('B') == 1) and (len(x) >= 20)]

print(len(array))
